In [6]:
from apyori import apriori
import pyfpgrowth
import numpy as np
import pandas as pd
import csv
from matplotlib import pyplot as plt

In [2]:
transactions = [[1, 2, 5],
                [2, 4],
                [2, 3],
                [1, 2, 4],
                [1, 3],
                [2, 3],
                [1, 3],
                [1, 2, 3, 5],
                [1, 2, 3]]

In [3]:
results = list(apriori(transactions))

In [4]:
results

[RelationRecord(items=frozenset({1}), support=0.6666666666666666, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({1}), confidence=0.6666666666666666, lift=1.0)]),
 RelationRecord(items=frozenset({2}), support=0.7777777777777778, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({2}), confidence=0.7777777777777778, lift=1.0)]),
 RelationRecord(items=frozenset({3}), support=0.6666666666666666, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({3}), confidence=0.6666666666666666, lift=1.0)]),
 RelationRecord(items=frozenset({4}), support=0.2222222222222222, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({4}), confidence=0.2222222222222222, lift=1.0)]),
 RelationRecord(items=frozenset({5}), support=0.2222222222222222, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({5}), confidence=0.2222222222222222, lift=1.0)]),
 RelationRecord

In [22]:
patterns = pyfpgrowth.find_frequent_patterns(transactions, 2)
rules = pyfpgrowth.generate_association_rules(patterns, 0.7)

In [23]:
rules

{(1, 5): ((2,), 1.0),
 (2, 5): ((1,), 1.0),
 (4,): ((2,), 1.0),
 (5,): ((1, 2), 1.0)}

In [1]:
patterns

NameError: name 'patterns' is not defined

# 결과비교
## Apriori
|item1|item2|support|confidence|lift|
|-|-|-|-|-|
|1|2|0.44444444|0.66666667|0.85714286|
|2|1|0.44444444|0.57142857|0.85714286|
|1|3|0.44444444|0.66666667|1.00000000|
|3|1|0.44444444|0.66666667|1.00000000|
|4|1|0.11111111|0.50000000|0.75000000|
|5|1|0.22222222|1.00000000|1.50000000|
|2|3|0.44444444|0.57142857|0.85714286|
|3|2|0.44444444|0.66666667|0.85714286|
|4|2|0.22222222|1.00000000|1.28571429|
|5|2|0.22222222|1.00000000|1.28571429|
|5|3|0.11111111|0.50000000|0.75000000|
## FPgrowth
* Minsupport = 2
* Minconficence = 0.7

|ancedent|consequence|confidence|
|-|-|-|
|1, 5|2    |1.0|
|2, 5|1    |1.0|
|4   |2    |1.0|
|5   |1, 2 |1.0|

# Test 2

In [23]:
transaction2 = [['Beer', 'Nuts', 'Diaper'],
                ['Beer', 'Coffee', 'Diaper'],
                ['Beer', 'Diaper', 'Eggs'],
                ['Nuts', 'Eggs', 'Milk'],
                ['Nuts', 'Coffee', 'Diaper', 'Eggs', 'Milk'],
                ['Coffee','Milk']]

In [24]:
results = list(apriori(transaction2))

In [25]:
results

[RelationRecord(items=frozenset({'Beer'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'Beer'}), confidence=0.5, lift=1.0)]),
 RelationRecord(items=frozenset({'Coffee'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'Coffee'}), confidence=0.5, lift=1.0)]),
 RelationRecord(items=frozenset({'Diaper'}), support=0.6666666666666666, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'Diaper'}), confidence=0.6666666666666666, lift=1.0)]),
 RelationRecord(items=frozenset({'Eggs'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'Eggs'}), confidence=0.5, lift=1.0)]),
 RelationRecord(items=frozenset({'Milk'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset(), items_add=frozenset({'Milk'}), confidence=0.5, lift=1.0)]),
 RelationRecord(items=frozenset({'Nuts'}), support=0.5, ordered_statistics=[O

In [28]:
patterns2 = pyfpgrowth.find_frequent_patterns(transaction2, 2)
rules2 = pyfpgrowth.generate_association_rules(patterns2, 0.5)

In [29]:
rules2

{('Beer',): (('Diaper',), 1.0),
 ('Coffee',): (('Milk',), 0.6666666666666666),
 ('Diaper',): (('Eggs',), 0.5),
 ('Eggs', 'Milk'): (('Nuts',), 1.0),
 ('Eggs', 'Nuts'): (('Milk',), 1.0),
 ('Milk', 'Nuts'): (('Eggs',), 1.0),
 ('Nuts',): (('Eggs', 'Milk'), 0.6666666666666666)}

# result
## apriori
|item1|item2|support|confidence|lift|
|-|-|-|-|-|
|Beer|Diaper|0.50000000|1.00000000|1.50000000|
|Diaper|Beer|0.50000000|0.75000000|1.50000000|
|Coffee|Diaper|0.33333333|0.66666667|1.00000000|
|Diaper|Coffee|0.33333333|0.50000000|1.00000000|
|Coffee|Milk|0.33333333|0.66666667|1.33333333|
|Milk|Coffee|0.33333333|0.66666667|1.33333333|
|Diaper|Eggs|0.33333333|0.50000000|1.00000000|
|Eggs|Diaper|0.33333333|0.66666667|1.00000000|
|Diaper|Nuts|0.33333333|0.50000000|1.00000000|
|Nuts|Diaper|0.33333333|0.66666667|1.00000000|
|Eggs|Milk|0.33333333|0.66666667|1.33333333|
|Milk|Eggs|0.33333333|0.66666667|1.33333333|
|Eggs|Nuts|0.33333333|0.66666667|1.33333333|
|Nuts|Eggs|0.33333333|0.66666667|1.33333333|
|Milk|Nuts|0.33333333|0.66666667|1.33333333|
|Nuts|Milk|0.33333333|0.66666667|1.33333333|
## FPgrowth
|item1|item2|confidence|
|-|-|-|
|Beer|Diaper|1.0|
|Coffee|Milk|0.6666666666666666|
|Diaper|Eggs|0.5|
|Eggs,Milk|Nuts|1.0|
|Eggs,Nuts|Milk|1.0|
|Milk,Nuts|Eggs|1.0|
|Nuts|Eggs,Milk|0.6666666666666666|